src/test/fuzz/addrman.cpp

```diff
@@ -25,25 +26,172 @@ void initialize_addrman()
 class CAddrManDeterministic : public CAddrMan
 {
 public:
-    void MakeDeterministic(const uint256& random_seed)
+    explicit CAddrManDeterministic(FuzzedDataProvider& fuzzed_data_provider)
     {
-        insecure_rand = FastRandomContext{random_seed};
-        Clear();
+        insecure_rand = FastRandomContext{ConsumeUInt256(fuzzed_data_provider)};
+        if (fuzzed_data_provider.ConsumeBool()) {
+            m_asmap = ConsumeRandomLengthBitVector(fuzzed_data_provider);
+            if (!SanityCheckASMap(m_asmap)) {
+                m_asmap.clear();
+            }
+        }
+    }
+
+    /**
+     * Generate a random address.
+     */
+    CNetAddr RandAddr()
+    {
+        // The networks [1..6] correspond to CNetAddr::BIP155Network (private).
+        static const std::map<uint8_t, uint8_t> net_len_map = {
+            {1, ADDR_IPV4_SIZE},  {2, ADDR_IPV6_SIZE}, {3, ADDR_TORV2_SIZE},
+            {4, ADDR_TORV3_SIZE}, {5, ADDR_I2P_SIZE},  {6, ADDR_CJDNS_SIZE}};
+        const uint8_t net = insecure_rand.randrange(6) + 1; // [1..6]
+
+        CDataStream s(SER_NETWORK, PROTOCOL_VERSION | ADDRV2_FORMAT);
+
+        s << net;
+        s << insecure_rand.randbytes(net_len_map.at(net));
```

Comment:
Wouldn't it be better to read the address from the fuzz input? Fuzz input isn't uniform and thus can easier reach edge-cases like `000000`, which would be non-trivial or next to impossible with a uniform distribution over a large enough range.


Also, how is this different from `ConsumeNetAddr`?

Reply:
The problem with consuming addresses directly from the fuzz input is that it is limited in size. This test needs 1000s of addresses. What happened in practice was that some random addresses were consumed followed by 1000s of all-zero addresses due to the fuzzing input being exhausted.

That is the main difference from `ConsumeNetAddr()`.

Reply:
Maybe add a few fuzzed addresses and fill up the rest with synthetic ones?

Reply:
Right. The fuzzed provider is not used for anything after consuming the pile of addresses, so it is ok to fully exhaust it. Once it is exhausted, start generating ones from `insecure_rand`.